templates/javascript.js.twig

```diff
@@ -102,3 +102,107 @@ $(document).ready(function() {
     removeSkill();
 });
 {% endmacro %}
+
+{% macro citizenProjectSkills(idSkillsListField, idSkillSearchField ,urlSearch) %}
+$(document).ready(function() {
+    var skillList = $('{{ idSkillsListField|raw }}');
+    var removeSkill = function () {
+        var btn = $('.skill-remove');
+
+        btn.unbind('click');
+        btn.click(function () {
+            $(this).parent().remove();
+        });
+    };
+
+    var addSkill = function (item) {
+        var skill = item.value,
+            label = item.label;
+
+        if (skill) {
+            var skillCount = 0;
+            $('.summary-skill > input[type=hidden]').each(function () {
+                skillCount = Math.max(this.id.replace(/[^0-9]/g, ''), skillCount);
+            });
+            skillCount++;
+
+            var newSkillWidget = skillList.attr('data-prototype'),
+                skillWidgetId = '#'+skillList.attr('id')+'_'+skillCount;
+            newSkillWidget = newSkillWidget.replace(/__name__/g, skillCount);
+
+            $(newSkillWidget).appendTo(skillList);
+            $(skillWidgetId).val(skill);
+            $(skillWidgetId).parent().find('.skill-value').text(' ' + label);
+
+            $('{{ idSkillSearchField|raw }}').val('');
+            removeSkill();
+        }
+    };
+
+    // Prevent submitting form by pressing enter
+    $(window).keydown(function (event) {
+        if (event.keyCode == 13) {
+            event.preventDefault();
+            return false;
+        }
+    });
+
+    var input = $('{{ idSkillSearchField|raw }}');
+
+    // Add skill autocomplete
+    input.autocomplete({
+        source: function(request, response) {
+            var term = request.term.trim(),
+                categoryId = $('#citizen_project_category').val();
+            if (term.length > 0 && categoryId) {
+                $.get("{{ urlSearch|raw }}?term=" + term+'&category='+categoryId, function(data) {
```

Comment:
We could save up some HTTP calls by wrapping the whole function in a `setTimeout` function.
See https://github.com/EnMarche/en-marche.fr/pull/1842/files#diff-abdc72dfc05d1f015d1308f8b7bae8cbR50